src/workerd/jsg/modules.h

```diff
@@ -395,9 +397,43 @@ class ModuleRegistryImpl final: public ModuleRegistry {
     for (auto module: bundle.getModules()) {
       auto type = module.getType();
       auto filter = maybeFilter.orDefault(type);
-      if (type == filter) {
+      if (type != filter) {
+        continue;
+      }
+      auto specifier = module.getName();
+      auto path = kj::Path::parse(specifier);
+      using Key = typename Entry::Key;
+      if (type == Type::BUILTIN && entries.find(Key(path, Type::BUNDLE)) != kj::none) {
+        continue;
+      }
+
+      switch (module.which()) {
+      case Module::SRC:
         // TODO: asChars() might be wrong for wide characters
-        addBuiltinModule(module.getName(), module.getSrc().asChars(), type);
+        addBuiltinModule(specifier, module.getSrc().asChars(), type);
+        break;
```

Comment:
All of these changes should be reverted or put behind a gate as well.

Reply:
I know it's a pain, but ideally we want to put all changes behind the gate and avoiding any code changes without the gate being enabled.

Might have to restructure this a bit to make that possible, maybe instead of adding a union, add a `srcType` field and make it default to `JS`. This way existing code can continue to work without changes.

Reply:
I can just revert the last two commits.

Reply:
I have two local branches of this: `wasm-builtin` which I made first, pushed to `hoodmane/wasm-builtin`, and made the PR from, and `hoodmane/wasm-builtin` which I pushed to `cloudflare/workerd` after I got rights to push stuff there. I seem to have reverted these commits only on `hoodmane/wasm-builtin`...